### ISSUE TITLE
When starting web service, if the keystore doesn't exist, there is a DEBUG instead of ERROR message

### ISSUE THREAD

**Issue by ligadata-william** [LigaData/Kamanja#723]
When I run ./kamanja start webservice, I get the following error because I haven't generated a keystore for it:
```
DEBUG [main] - Stacktrace:java.io.IOException: Could not open /tmp/Kamanja/config/keystore.jks as a file, class path resource, or URL.
	at org.apache.camel.util.jsse.JsseParameters.resolveResource(JsseParameters.java:166)
	at org.apache.camel.util.jsse.KeyStoreParameters.createKeyStore(KeyStoreParameters.java:171)
	at org.apache.camel.util.jsse.KeyManagersParameters.createKeyManagers(KeyManagersParameters.java:106)
	at org.apache.camel.util.jsse.SSLContextParameters.createSSLContext(SSLContextParameters.java:234)
	at com.ligadata.metadataapiservice.LigadataSSLConfiguration$class.sslContext(LigadataSSLConfiguration.scala:31)
	at com.ligadata.metadataapiservice.APIService.sslContext(APIService.scala:20)
	at com.ligadata.metadataapiservice.APIService.com$ligadata$metadataapiservice$APIService$$StartService(APIService.scala:128)
	at com.ligadata.metadataapiservice.APIService$.main(APIService.scala:156)
	at com.ligadata.metadataapiservice.APIService.main(APIService.scala)
```

This stack trace is written as a DEBUG message but should be an ERROR message.

Additionally, we should handle this differently in one of two ways:

My preferred method would be to check if that file "/tmp/Kamanja/config/keystore.jks" exists first and if it doesn't, give an ERROR message and return an error (code?) that should ultimately result in the process ending.

The other way would be to catch java.io.IOException and give an ERROR message there and exit out the same way as described in the above method.

Either way, any exceptions we catch should result in an ERROR message and a shutting down of the process.

**Comment by ligadata-dhaval** [comment]
As william mentioned, Kamanja web service thread does not stop when errors occur. Probably because it has Thread.sleep(3652460601000L). We need to handle all generic exceptions and close the web service when any exception occurs. I have closed the issue 1191as duplicate which is on similar lines as this.

**Comment by ligadata-dhaval** [comment]
Hi Yasser, 
Below is my analysis on the web service:

system.shutdown should shutdown the akka actor system which is the web service for us. This will shut the web service down. However, the zkListener is creating curator clients which stay connected to the zkServer. Even if we say client.close, it causes the clients to go into suspended state instead of LOST. Due to this, it reconnects to the server as per the retry logic.

**Comment by ligadata-yasser** [comment]
Hi Dhaval,
on branch (1.5.0-Kamanja-583_Kamanja-723), i added system.shutdown to service Shutdown() method, and also added MetadataAPIImpl.shutdown. 
so far it worked because in the first tests i made, if there is something wrong with config (like missing password), the service just showed a warning and stopped - which is exactly what i want.
the only problem i am facing now, is when the service runs successfully and then i try to stop it (Ctrl+C), it gives an exception (java.lang.IllegalAccessError: already closed) as if it is trying to close store twice. i don't think this is because of my changes, since the issue was even before i did them (i got the same issue while testing other branches that don't have my changes).
sometimes instead of showing this exception, it shows a jvm fatal error.
There is where i am stuck now

**Comment by ligadata-yasser** [comment]
@ligadata-william 

I made a few fixes:
- if ssl is disabled, no need for certificate file or passowrd, the service runs http normally
- if ssl is enabled, check for SSL_CERTIFICATE and SSL_PASSWD values, if one of them is missing or certificate file does not exist, give an error msg and shutdown service completely
- if passowrd is provided but was not correct an exceptin is shown as (ERROR) and service is shutdown completely

these fixes are for the following issues: #567, #583, #723, #724
please verify (and if you can verify code changes), pull request is #1195

now there is one left issue: an exception is occasionally thrown when shutting down the service (CTRL+C) if storage is hashmap. i will open a separate issue for it.

**Comment by ligadata-william** [comment]
Verified this is fixed. Closing.
